MediaBrowser.Api/PluginService.cs

```diff
@@ -243,16 +245,16 @@ public async Task Post(UpdatePluginConfiguration request)
             // https://code.google.com/p/servicestack/source/browse/trunk/Common/ServiceStack.Text/ServiceStack.Text/Controller/PathInfo.cs
             var id = Guid.Parse(GetPathValue(1));
 
-            var plugin = _appHost.Plugins.First(p => p.Id == id) as IHasPluginConfiguration;
+            var plugin = _appHost.Plugins.FirstOrDefault(p => p.Id == id);
```

Comment:
Not the same behaviour

Reply:
Well i modified it becouse here if no plugin satisfies the condition an InvalidOperationException is thrown. Then the if below tests whether the plugin is null and if so then throws a FileNotFoundException.

Here the desired behavior would seem to be that when no plugin is found then the method should throw a FileNotFoundException.

The version with FirstOrDefault will not throw InvalidOperationException. Should this method throw this exception?

Reply:
done